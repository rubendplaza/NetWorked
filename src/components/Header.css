.navBar {
  background-color: #1f2833;
  width: 100%;
}

.main-nav {
  display: flex;
  list-style: none;
  font-size: 1.5em;
  margin: 0;
  align-items: center;
}

@media only screen and (max-width: 600px) {
  .main-nav {
    font-size: 1em;
    padding: 0;
  }
}

.linkItem {
  padding: 20px;
}

.link {
  font-size: 1.2em;
  text-decoration: none;
  color: #ffffff;
}

.link:hover {
  color: #66fcf1;
}

.push {
  margin-left: auto;
}

.appName {
  font-size: 2em;
  color: #66fcf1;
  font-weight: 700;
}
