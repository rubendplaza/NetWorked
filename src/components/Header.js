import React from 'react';
import './Header.css';

const Header = () => {
    return (
        <nav className='navBar sticky'>
            <ul className='main-nav'>
                <li className='linkItem'><a href='google.com' className='link appName'>NetWorked</a></li>
                <li className='linkItem push'><a href='google.com' className='link'>About</a></li>
                <li className='linkItem'><a href='google.com' className='link'>Contact</a></li>
                <li className='linkItem'><a href='google.com' className='link'>Team</a></li>
            </ul>
        </nav>
    )
}

export default Header;