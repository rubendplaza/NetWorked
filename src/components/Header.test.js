import React from "react";
import Header from "./Header";
import { shallow } from "enzyme";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Header />);
});

describe("testing Header component", () => {
  test("Header component renders correctly without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("tests shouldComponentUpdate returning false", () => {
    const mockProps = {};
    const mockState = {};
    const shouldUpdate = wrapper
      .instance()
      .shouldComponentUpdate(mockProps, mockState);
    expect(shouldUpdate).toBe(false);
  });
});
