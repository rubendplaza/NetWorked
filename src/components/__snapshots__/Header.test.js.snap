// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header component Header component renders correctly without crashing 1`] = `
<nav
  className="navBar sticky"
>
  <ul
    className="main-nav"
  >
    <li
      className="linkItem"
    >
      <a
        className="link appName"
        href="https://www.google.com/"
      >
        NetWorked
      </a>
    </li>
    <li
      className="linkItem push"
    >
      <a
        className="link"
        href="https://www.google.com/"
      >
        About
      </a>
    </li>
    <li
      className="linkItem"
    >
      <a
        className="link"
        href="https://www.google.com/"
      >
        Contact
      </a>
    </li>
    <li
      className="linkItem"
    >
      <a
        className="link"
        href="https://www.google.com/"
      >
        Team
      </a>
    </li>
  </ul>
</nav>
`;
